### DISCUSSION TITLE
useCallback with dynamic import component

### DISCUSSION THREAD

**Discussion by IRediTOTO** [vercel/next.js#14334] [Help]
Hi, I have a component Facebook comment like this
```
const Comment = dynamic(() =>
   import("../../../lib/facebook/comments-facebook"),{ssr:false}
)
```
it work fine
it's inside a tab, so when I click to the tab, it re-render, this look not good
i tried to use useCallback like this
`const CommentElement= useCallback(()=><Comment />)`
But error... Did I do something wrong?
![image](https://user-images.githubusercontent.com/15695252/85066269-d43c5f00-b1d8-11ea-8bc8-b91a591f6b05.png)

**Comment by jamesmosier** [comment]
`useCallback` isn't meant for rendering components. I don't think this has anything to do with next/dynamic though. `next/dynamic` just helps to render the component in a separate bundle. After the initial load, next/dynamic doesn't do anything.

Also dynamic components must be outside of the function body. See this issue for more info https://github.com/vercel/next.js/issues/13711

Aside from that, do you have any other code you could share? It is hard to know where the issue is within your component without seeing it.

**Comment by IRediTOTO** [comment]
Thank you for replied me @jaredpalmer  :)
I have a component, with load Facebook comment and Facebook script
import { useEffect, memo } from "react";

```
const CommentsFacebook =memo((props) => { // 
   console.log("render CommentsFacebook")
   useEffect(() => {
      console.log("load facebook script")
      const facebookScript = document.createElement("script");
      facebookScript.async = true;
      facebookScript.src = `https://connect.facebook.net/en_US/sdk.js#xfbml=1&version=v7.0&appId=${process.env.FACEBOOK_APP_ID}&autoLogAppEvents=1`;
      document.body.appendChild(facebookScript);
   },[]);

   return (
      <React.Fragment>
         <div id="fb-root"></div>
         <div className="fb-comments" data-href={props.fullUrl} data-numposts="10" data-width="100%"></div>
      </React.Fragment>
   );
});
export default CommentsFacebook;
```
Then I load it use dynamic import in product page, I just want FacebookComment don't re-render when I click to a tab. 
```
const Comment = dynamic(
   () => import("../../../lib/facebook/comments-facebook"),
   { ssr: false }
);
```
```
<div className={tabClick == 3 ? "box" : "is-hidden"}>
                     <h3 className="has-text-weight-bold is-size-3">Comment</h3>
                   
                     <Comment fullUrl={process.env.WEB_URI + router.asPath} />
                  </div>
```
Those code above was solved my re-render problem :) **I used React.memo** . Just wonder is there a better way.
